lib/twingly/http.rb

```diff
@@ -44,9 +44,11 @@ class Client # rubocop:disable Metrics/ClassLength
       attr_accessor :follow_redirects_limit
       attr_accessor :retryable_exceptions
 
-      def initialize(logger:, base_user_agent:)
-        @logger          = logger
+      def initialize(base_user_agent:, logger: nil)
         @base_user_agent = base_user_agent
+        @logger = logger || begin
```

Comment:
I don't think you need the `begin`/`end`?

```suggestion
        @logger = logger || Logger.new("/dev/null")
```

Also, you could do it as, `def initialize(base_user_agent:, logger: Logger.new("/dev/null")`, but even better could be something like:

```
def initialize(base_user_agent:, logger: default_logger)
...
end

private 

def default_logger
  Logger.new("/dev/null")
end
```

Reply:
> I don't think you need the begin/end?

I realize now that you're absolutely right!

> Also, you could do it as, def initialize(base_user_agent:, logger: Logger.new("/dev/null"), but even better could be something like...

Didn't want to put it in the definition like that, but I really like the final alternative! :)

Reply:
Updated accordingly in a7d3d7e